src/digital.jl

```diff
@@ -8,8 +8,8 @@ for (cfunction, jfunction, ntask) in (
     end
     @eval function $jfunction(t::$ntask, channel::String)
         catch_error( $cfunction(t.th,
-                Ref(codeunits(channel),1),
-                Ref(codeunits(""),1),
+                pointer(channel),
+                pointer(""),
```

Comment:
in #17 you hypothesized that the the error when trying to add `ai10` was due to the `Ref` here, and that changing to `pointer` fixed it.  but is that really the case, or was using `NRSE` instead of `Differential` what was needed?  i ask, because `Ref` is "generally preferred" to `pointer` according to the latter's docstring.

Reply:
I was using the 144 analog input card. So it was kind of weird to be stopped after so few. I think the reason it was breaking was between the code units and ref. I think keeping these as pointers is safe in the context of the function call since they aren't being modified, or even declared in the function body. They shouldn't be garbage collected in the middle of a function call since they're arguments. But that's me speaking out of ignorance to how exactly the GC works.

I vote for keeping these specifically as pointers due to the aforementioned reasons.

Reply:
@Cody-G experienced intermittent [problems](https://github.com/JaneliaSciComp/NIDAQ.jl/pull/15#issuecomment-457632258) with `pointer` which were fixed by changing to `Ref`.

would be nice to know for sure what your problem was before changing this.

Reply:
Interesting results. It seems only a certain range of values fail:
```julia> t0 = analog_input("Dev1/ai0", terminal_config=Differential)
NIDAQ.AITask(Ptr{Nothing} @0x000000001f20f900)

julia> t1 = analog_input("Dev1/ai10", terminal_config=Differential)
ERROR: NIDAQmx: Requested value is not a supported value for this property. The property value may be invalid because it conflicts with another property.
Stacktrace:
 [1] error(::String) at .\error.jl:33
 [2] #catch_error#1(::typeof(error), ::Function, ::Int32, ::String) at C:\Users\aporter\.julia\dev\NIDAQ\src\NIDAQ.jl:104
 [3] catch_error at C:\Users\aporter\.julia\dev\NIDAQ\src\NIDAQ.jl:97 [inlined] (repeats 2 times)
 [4] #analog_input#5(::TerminalConfig, ::Nothing, ::Function, ::NIDAQ.AITask, ::String) at C:\Users\aporter\.julia\dev\NIDAQ\src\analog.jl:23
 [5] #analog_input#4 at .\none:0 [inlined]
 [6] (::getfield(NIDAQ, Symbol("#kw##analog_input")))(::NamedTuple{(:terminal_config,),Tuple{TerminalConfig}}, ::typeof(analog_input), ::String) at .\none:0
 [7] top-level scope at none:0

julia> t2 = analog_input("Dev1/ai20", terminal_config=Differential)
NIDAQ.AITask(Ptr{Nothing} @0x000000001f20fc70)
```
Here is the result of me going from 0 - 143. Sorry for the bloat. I can't seem to upload files to comments.
```
julia> for i=0:143
           try
               t = analog_input("Dev1/ai$i", terminal_config=Differential)
               println("$i passed")
           catch
               println("$i failed")
           end
       end
0 passed
1 passed
2 passed
3 passed
4 passed
5 passed
6 passed
7 passed
8 failed
9 failed
10 failed
11 failed
12 failed
13 failed
14 failed
15 failed
16 passed
17 passed
18 passed
19 passed
20 passed
21 passed
22 passed
23 passed
24 failed
25 failed
26 failed
27 failed
28 failed
29 failed
30 failed
31 failed
32 passed
33 passed
34 passed
35 passed
36 passed
37 passed
38 passed
39 passed
40 failed
41 failed
42 failed
43 failed
44 failed
45 failed
46 failed
47 failed
48 passed
49 passed
50 passed
51 passed
52 passed
53 passed
54 passed
55 passed
56 failed
57 failed
58 failed
59 failed
60 failed
61 failed
62 failed
63 failed
64 passed
65 passed
66 passed
67 passed
68 passed
69 passed
70 passed
71 passed
72 failed
73 failed
74 failed
75 failed
76 failed
77 failed
78 failed
79 failed
80 passed
81 passed
82 passed
83 passed
84 passed
85 passed
86 passed
87 passed
88 failed
89 failed
90 failed
91 failed
92 failed
93 failed
94 failed
95 failed
96 passed
97 passed
98 passed
99 passed
100 passed
101 passed
102 passed
103 passed
104 failed
105 failed
106 failed
107 failed
108 failed
109 failed
110 failed
111 failed
112 passed
113 passed
114 passed
115 passed
116 passed
117 passed
118 passed
119 passed
120 failed
121 failed
122 failed
123 failed
124 failed
125 failed
126 failed
127 failed
128 passed
129 passed
130 passed
131 passed
132 passed
133 passed
134 passed
135 passed
136 failed
137 failed
138 failed
139 failed
140 failed
141 failed
142 failed
143 failed
```

Is there something I am missing?

Reply:
Here we go:
![image](https://user-images.githubusercontent.com/9441838/55351761-021e7200-5474-11e9-955f-2c9be6f34f99.png)
I am attempting to use invalid configurations. I will roll back to using Ref.